packages/alfa-rules/src/sia-r65/rule.ts

```diff
@@ -25,10 +28,53 @@ export default Rule.Atomic.of<Page, Element, Question.Metadata>({
   requirements: [Criterion.of("2.4.7")],
   tags: [Scope.Component],
   evaluate({ device, document }) {
+    let diagnostic = Map.empty<string, Matches>();
+
+    const tabbables = document.tabOrder().filter(isTabbable(device));
+    const nonTargets = document
+      .descendants()
+      .filter((node) => isElement(node) && !tabbables.includes(node));
+    const targetClassnames = tabbables.flatMap((tabbable) => tabbable.classes);
+
+    let nonTargetClassnames = Sequence.empty<string>();
+    for (const nonTarget of nonTargets) {
+      if (isElement(nonTarget)) {
```

Comment:
This `isElement` test shouldn't be needed because `nonTargets` has already be filtered. The `&&` should be replace by `Refinement.and` to keep the type narrowing. Then `nonTargetClassnames` can probably be defined exactly the same way as `targetClassnames`.